src/System.Globalization.Calendars/tests/CalendarHelpers.cs

```diff
@@ -2,177 +2,413 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Generic;
 using Xunit;
 
 namespace System.Globalization.Tests
 {
     public static class CalendarHelpers
     {
-        private static int MaxEra(Calendar calendar) => calendar.GetEra(calendar.MaxSupportedDateTime);
+        public enum DataType
+        {
+            Year = 1,
+            Month = 2,
+            Day = 8
+        }
+        
+        public static Calendar[] s_calendars = new Calendar[]
+        {
+            new ThaiBuddhistCalendar(),
+            new JapaneseCalendar(),
+            new KoreanCalendar(),
+            new TaiwanCalendar(),
+            new GregorianCalendar(GregorianCalendarTypes.Arabic),
+            new GregorianCalendar(GregorianCalendarTypes.Localized),
+            new GregorianCalendar(GregorianCalendarTypes.MiddleEastFrench),
+            new GregorianCalendar(GregorianCalendarTypes.TransliteratedEnglish),
+            new GregorianCalendar(GregorianCalendarTypes.TransliteratedFrench),
+            new GregorianCalendar(GregorianCalendarTypes.USEnglish),
+            new HijriCalendar(),
+            new HebrewCalendar(),
+            new JulianCalendar(),
+            new TaiwanLunisolarCalendar(),
+            new ChineseLunisolarCalendar(),
+            new KoreanLunisolarCalendar(),
+            new PersianCalendar()
+        };
+
         private static int MinEra(Calendar calendar) => calendar.GetEra(calendar.MinSupportedDateTime);
+        private static int MaxEra(Calendar calendar) => calendar.GetEra(calendar.MaxSupportedDateTime);
 
-        private static int YearOffset(Calendar calendar) => calendar.GetYear(calendar.MinSupportedDateTime) - 1;
+        private static int MaxCalendarYearInEra(Calendar calendar, int era)
+        {
+            int[] eras = calendar.Eras;
+            Assert.InRange(era, 0, eras[0]);
+            if (eras.Length == 1 || era == eras[0] || era == 0)
+            {
+                return calendar.GetYear(calendar.MaxSupportedDateTime);
+            }
+            return MinGregorianYearInEra(calendar, era + 1) - MinGregorianYearInEra(calendar, era) + 1;
```

Comment:
one last comment, this line doesn't look right to me. this line will return the difference in years between the minimum years of 2 eras which is not really the max year of the requested era. this may need to be 

return cal.GetYear(cal.ToDateTime(1, 1, 1, 0, 0, 0, 0, era + 1).AddDays(-1))